ros2topic/ros2topic/api/__init__.py

```diff
@@ -146,12 +155,19 @@ def __call__(self, prefix, parsed_args, **kwargs):
 
 def qos_profile_from_short_keys(
     preset_profile: str, reliability: str = None, durability: str = None,
+    depth: int = -1, history: str = None,
 ) -> rclpy.qos.QoSProfile:
     """Construct a QoSProfile given the name of a preset, and optional overrides."""
     # Build a QoS profile based on user-supplied arguments
     profile = rclpy.qos.QoSPresetProfiles.get_from_short_key(preset_profile)
+    if depth >= 0:
+        profile.depth = depth
+    if history:
+        profile.history = rclpy.qos.QoSHistoryPolicy.get_from_short_key(history)
     if durability:
         profile.durability = rclpy.qos.QoSDurabilityPolicy.get_from_short_key(durability)
+        if profile.durability == rclpy.qos.QoSDurabilityPolicy.TRANSIENT_LOCAL and profile.depth == 0:
+            profile.depth = 1
     if reliability:
         profile.reliability = rclpy.qos.QoSReliabilityPolicy.get_from_short_key(reliability)
```

Comment:
I just realized this has the same problem of overriding the users' preference.  That is, if the user says `--qos-durability transient_local --qos-depth 0`, then we'll automatically override to a depth of 1.

Arguably, that's a somewhat silly thing to do, but it is still something I think we should support.  I think you can fix this by rearranging this whole block to be something more like this:

```
    if history:
        profile.history = rclpy.qos.QoSHistoryPolicy.get_from_short_key(history)
    if durability:
        profile.durability = rclpy.qos.QoSDurabilityPolicy.get_from_short_key(durability)
    if reliability:
        profile.reliability = rclpy.qos.QoSReliabilityPolicy.get_from_short_key(reliability)
    if depth >= 0:
        profile.depth = depth
    else:
        if profile.durability == rclpy.qos.QoSDurabilityPolicy.TRANSIENT_LOCAL:
            profile.depth = 1
```

Reply:
that makes sense to me, i think it is better for user. appreciate for your considering!